common/exception/src/exception.rs

```diff
@@ -75,12 +75,16 @@ impl ErrorCode {
     pub fn add_message_back(self, msg: impl AsRef<str>) -> Self {
         Self {
             code: self.code(),
-            display_text: format!("{}{}", self.display_text, msg.as_ref()),
+            display_text: format!("{}{}", self.rewirte_err_msg_for_groupby(&self.display_text), msg.as_ref()),
             cause: self.cause,
             backtrace: self.backtrace,
         }
     }
 
+    pub fn rewirte_err_msg_for_groupby(&self, display_text: &String) -> String{
+        display_text.replace("\"", "`").replace("Unable to get field named", "Column").replace(". Valid fields:","is not under aggregate function and not in GROUP BY: While processing ")
```

Comment:
Hi, it would be better that we check and return the error from the underlying, instead of replaced.

Reply:
For issue #3823 , we don't need to follow the ClickHouse way, but the error is not correct for that case, it looks we can't only replace the message text.

Reply:
> For issue #3823 , we don't need to follow the ClickHouse way, but the error is not correct for that case, it looks we can't only replace the message text.

Give more info.

Reply:
Thanks for reminding,  I think the issue #3823 should be closed(I have closed), it's not a critical error, from the initial plan of this issue, we want to get a better way to make the message looks more better, it looks there is no good way.
Feel free to comment.

Reply:
> Hi, it would be better that we check and return the error from the underlying, instead of replaced.
Yes, fn add_message_back() is so fundamental.
I think the original error message is good enough to express the error, we may use this kind of field search in many places (1 to n model) . Maybe the position to make the message more specific for each case is the like 

`               Err(cause) => {
                    return Err(cause.add_message_back(" (while in select before group by)"));
}`
give a cause some pub method to change its information in every case. 
Maybe that's ok.